File: dart/lib/api/isochrone_api.dart
part of openapi.api;



class IsochroneApi {
  final ApiClient apiClient;

  IsochroneApi([ApiClient apiClient]) : apiClient = apiClient ?? defaultApiClient;

  /// Isochrone Request
  ///
  /// The GraphHopper Isochrone API allows calculating an isochrone of a locations means to calculate &#39;a line connecting points at which a vehicle arrives at the same time,&#39; see [Wikipedia](http://en.wikipedia.org/wiki/Isochrone_map). It is also called **reachability** or **walkability**. 
  Future<IsochroneResponse> isochroneGet(String point, String key, { int timeLimit, int distanceLimit, String vehicle, int buckets, bool reverseFlow, String weighting }) async {
    Object postBody = null;

    // verify required params are set
    if(point == null) {
     throw new ApiException(400, "Missing required param: point");
    }
    if(key == null) {
     throw new ApiException(400, "Missing required param: key");
    }

    // create path and map variables
    String path = "/isochrone".replaceAll("{format}","json");

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
      queryParams.addAll(_convertParametersForCollectionFormat("", "point", point));
    if(timeLimit != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "time_limit", timeLimit));
    }
    if(distanceLimit != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "distance_limit", distanceLimit));
    }
    if(vehicle != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "vehicle", vehicle));
    }
    if(buckets != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "buckets", buckets));
    }
    if(reverseFlow != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "reverse_flow", reverseFlow));
    }
    if(weighting != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "weighting", weighting));
    }
      queryParams.addAll(_convertParametersForCollectionFormat("", "key", key));

    List<String> contentTypes = [];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
    }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'IsochroneResponse') as IsochroneResponse;
    } else {
      return null;
    }
  }
}

File: ruby-sinatra/api/isochrone_api.rb
require 'json'


MyApp.add_route('GET', '/api/1/isochrone', {
  "resourcePath" => "/Isochrone",
  "summary" => "Isochrone Request",
  "nickname" => "isochrone_get", 
  "responseClass" => "IsochroneResponse", 
  "endpoint" => "/isochrone", 
  "notes" => "The GraphHopper Isochrone API allows calculating an isochrone of a locations means to calculate 'a line connecting points at which a vehicle arrives at the same time,' see [Wikipedia](http://en.wikipedia.org/wiki/Isochrone_map). It is also called **reachability** or **walkability**. ",
  "parameters" => [
    {
      "name" => "point",
      "description" => "Specify the start coordinate",
      "dataType" => "string",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "time_limit",
      "description" => "Specify which time the vehicle should travel. In seconds.",
      "dataType" => "int",
      "allowableValues" => "",
      "defaultValue" => "null",
      "paramType" => "query",
    },
    {
      "name" => "distance_limit",
      "description" => "Specify which distance the vehicle should travel. In meter.",
      "dataType" => "int",
      "allowableValues" => "",
      "defaultValue" => "null",
      "paramType" => "query",
    },
    {
      "name" => "vehicle",
      "description" => "Possible vehicles are bike, car, foot and [more](https://graphhopper.com/api/1/docs/supported-vehicle-profiles/)",
      "dataType" => "string",
      "allowableValues" => "",
      "defaultValue" => "null",
      "paramType" => "query",
    },
    {
      "name" => "buckets",
      "description" => "For how many sub intervals an additional polygon should be calculated.",
      "dataType" => "int",
      "allowableValues" => "",
      "defaultValue" => "null",
      "paramType" => "query",
    },
    {
      "name" => "reverse_flow",
      "description" => "If &#x60;false&#x60; the flow goes from point to the polygon, if &#x60;true&#x60; the flow goes from the polygon \&quot;inside\&quot; to the point. Example usage for &#x60;false&#x60;&amp;#58; *How many potential customer can be reached within 30min travel time from your store* vs. &#x60;true&#x60;&amp;#58; *How many customers can reach your store within 30min travel time.*",
      "dataType" => "boolean",
      "allowableValues" => "",
      "defaultValue" => "null",
      "paramType" => "query",
    },
    {
      "name" => "weighting",
      "description" => "Can be fastest or shortest",
      "dataType" => "string",
      "allowableValues" => "",
      "defaultValue" => "null",
      "paramType" => "query",
    },
    {
      "name" => "key",
      "description" => "Get your key at graphhopper.com",
      "dataType" => "string",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

